### ISSUE TITLE
Questions about using the model

### ISSUE THREAD

**Issue by XxxAtlantis** [pnborchert/MultiRep#2]
If I want to use my own data for practical applications with this model, how should I organize the dataset?

**Comment by pnborchert** [comment]
You can use the same dataloaders by following the structure of the FewRel dataset (https://github.com/thunlp/FewRel/blob/master/data/train_wiki.json). For instance, `train_wiki.json` is structured as a dictionary of relation types ['P931', 'P4552', 'P140', ...], which contain a list of instances. Instances are a dictionary of `tokens` (comma separated list of words), `h` (head entity, list containing the string, Wikidata identifier, start token index, end token index) and `t` (tail entity, list containing the name, Wikidata identifier, start token index, end token index). The wikidata identifiers are not used, so you can insert a placeholder value.

**Example Instance:**
{'tokens': ['Merpati', 'flight', '106', 'departed', 'Jakarta', '(', 'CGK', ')', 'on', 'a', 'domestic',  'flight', 'to',  'Tanjung',  'Pandan',  '(',  'TJQ', ')', '.'], 'h': ['tjq', 'Q1331049', [[16]]], 't': ['tanjung pandan', 'Q3056359', [[13, 14]]]}

**Comment by XxxAtlantis** [comment]
Thank you for your reply. I have some questions about the dataset used for this task. In the FewRel dataset, the training and validation sets have 700 examples for each relation type, while the test set is defined by four scenarios ('5 way 1 shot', '5 way 5 shot', '10 way 1 shot', '10 way 5 shot'). In practice, I don't have 700 examples for each relation type like in the training and validation sets. How should I apply this model in a low-resource scenario? Please point out if my understanding is incorrect.

**Comment by pnborchert** [comment]
In case you have very limited data, you can try few-shot domain adaptation. We published a dataset that works well for training models in out-of-domain scenarios: https://github.com/pnborchert/CORE. Alternatively, you could mix your training data with existing datasets that contain text and relations within the same domain.
